### ISSUE TITLE
Locations Sharing not able to find exact location

### ISSUE THREAD

**Issue by hermann-san** [iNPUTmice/Conversations#4453]
Hi,
when I want to share my location in Conversations, it doesn't show me the correct location. It show's me some location that is 1 kilometer away.
I had the same problem with OSMand and I was able to fix it by changing  the "location source" (in the OSMand settings)  from "Google Play Services" to "Android API"
However, this setting doesn't appear to be available in Conversation.
GPS works normally and functions well in other apps like Magic Earth or Komoot.

I'm on  /e/OS with MicroG
Has somebody an idea how to fix this?

Thanks

**Comment by licaon-kter** [comment]
Osmand is installed from F-Droid?

I ask because to be able to use Google location you need to use Google proprietary libs, which Conversations does not use, and any app in F-Droid won't include to begin with.

Did you enable "use precise location" in Android Settings Apps Conversations Permissions Location?

**Comment by licaon-kter** [comment]
@hermann-san this should have been posted in the proper place here instead: https://codeberg.org/iNPUTmice/Conversations/issues

Did you read https://github.com/iNPUTmice/Conversations/ ? :)

**Comment by hermann-san** [comment]
yes, I've read that , and forgot about it . Sorry about that. Next time I'll remember it.

when I go to  Android Settings Apps Conversations Permissions Location (Android 11), it only says - access permitted during usage of app.  When I go to "all permissions" for Conversations, it says there that approximate and precise location is permitted in the foreground.
Also this is not something I can enable or disable. It just says that that it's permitted.

I've installed OSMand via Aurora App Store.

`to be able to use Google location you need to use Google proprietary libs`
/e/OS is supposed to be a google free OS. It uses MicroG as a replacement for Google Play Services. So I suppose there is no "Google location" on my phone.

I'm not sure what you mean with "GPS test".  I did test GPS with the various apps like Magic Earth, Komoot and others. It worked fine.  But also with Satstat, which also worked fine.and shows the found 9-11 satellites and the exact location on the map.

This problem is present on 2 different phone's, same model (both /e/OS with 2 different versions)

**Comment by licaon-kter** [comment]
GPS Test: https://f-droid.org/packages/com.android.gpstest.osmdroid

> I've installed OSMand via Aurora App Store.

So the version with Google libs.

Not sure what microG provides (if) for "google location".

But to get faster location I recommend (if still on microG 0.2.27) to install https://f-droid.org/packages/helium314.localbackend and https://f-droid.org/packages/org.fitchfamily.android.wifi_backend/ and https://f-droid.org/packages/org.fitchfamily.android.gsmlocation/ (select country and network)

microG 0.2.28 does not support plugins anymore.

My experience on Android 11 was always fine, location is rather fast and precise. But that depends a lot on device, location (if indoor etc).

**Comment by hermann-san** [comment]
Thanks for your suggestions. I've installed all 3 apps and it has indeed helped. Thanks for that.
I assume that the location was found via GSM before, that's why it was so bad and far away.
I'm not sure how the locations was found today. It was probably via WLAN and that might be a problem if I'm somewhere were there isn't WLAN.  
Also I'm on MicroG 0.2.27 and this may get updated with a newer version via a OS Upgrade at some point.
In any case, for now it works. Thanks again!
